packages/dds/tree/src/test/memory/tableTree.spec.ts

```diff
@@ -0,0 +1,685 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "node:assert";
+import {
+	benchmarkMemory,
+	isInPerformanceTestingMode,
+	type IMemoryTestObject,
+} from "@fluid-tools/benchmark";
+
+import {
+	Column,
+	Row,
+	UndoRedoManager,
+	createTableTree,
+	removeColumnAndCells,
+	type TableTreeDefinition,
+} from "../tablePerformanceTestUtilities.js";
+
+/**
+ * Note: These benchmarks are designed to closely match the benchmarks in SharedMatrix.
+ * If you modify or add tests here, consider updating the corresponding SharedMatrix benchmarks as well
+ * to ensure consistency and comparability between the two implementations.
+ */
+
+/**
+ * Creates a benchmark for undo operations on a SharedTree.
+ */
+function createUndoBenchmark({
+	title,
+	tableSize,
+	initialValue,
+	operationCount,
+	stackCount,
+	operation,
+}: {
+	title: string;
+	tableSize: number;
+	initialValue: string;
+	operationCount: number;
+	stackCount: number;
```

Comment:
It looks like all of the callers to this function are providing the same value for `operationCount` and `stackCount`, which doesn't seem right. For operations that do multiple edits, `stackCount` should presumably be different. I'm surprised that your `assert(!this.undoRedoManager.canUndo);` is passing. This may need some investigation.

That said, I'm also not sure it makes sense to have the caller provide the full `stackCount` here. Instead, they could specify the number of edits made per "operation". Then this function could calculate `stackCount` as `operationCount * editsPerOperation`.